SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -280,4 +281,177 @@ bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIs
+	}
+
+	FPlatformProcess::ClosePipe(0, ReadPipe);
+	FPlatformProcess::ClosePipe(0, WritePipe);
+
+	return ProcHandle;
+}
+
+void SpatialCommandUtils::StopLocalReceptionistProxyServer(FProcHandle& ProcHandle)
+{
+	if (ProcHandle.IsValid())
+	{
+		FPlatformProcess::TerminateProc(ProcHandle, true);
+	}
+}
+
+bool SpatialCommandUtils::GetProcessInfoFromPort(int32 Port, FString& OutPid, FString& OutState, FString& OutProcessName)
+{
+	bool bSuccess = false;
+#if PLATFORM_WINDOWS
+	bSuccess = GetProcessInfoFromPortWindows(Port, OutPid, OutState);
+#endif
+
+#if PLATFORM_MAC
+	bSuccess = GetProcessInfoFromPortMacOs(Port, OutPid, OutState, OutProcessName);
+#endif
+
+	return bSuccess;
+}
+
+bool SpatialCommandUtils::TryKillProcessWithPID(const FString& PID)
+{
+	int32 ExitCode;
+	FString StdErr;
+
+#if PLATFORM_WINDOWS
+	const FString KillCmd = TEXT("taskkill");
+	const FString KillArgs = FString::Printf(TEXT("/F /PID %s"), *PID);
+#endif
+
+#if PLATFORM_MAC
+	const FString KillCmd = FPaths::Combine(SpatialGDKServicesConstants::KillCmdFilePath, TEXT("kill"));
+	const FString KillArgs = FString::Printf(TEXT("%s"), *PID);
+#endif
+
+	FString KillResult;
+	bool bSuccess = FPlatformProcess::ExecProcess(*KillCmd, *KillArgs, &ExitCode, &KillResult, &StdErr);
+	bSuccess = bSuccess && ExitCode == 0;
+	if (!bSuccess)
+	{
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("Failed to kill process with PID %s. Error: %s"), *PID, *StdErr);
+	}
+
+	return bSuccess;
+}
+
+bool SpatialCommandUtils::GetProcessInfoFromPortWindows(int32 Port, FString& OutPid, FString& OutState)
+{
+	bool bSuccess = false;
+
+	const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString NetStatArgs = TEXT("-n -o -a");
+	FString NetStatResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*NetStatCmd, *NetStatArgs, &ExitCode, &NetStatResult, &StdErr);
+
+	if (ExitCode == 0 && bSuccess)
+	{
+		// Get the line of the netstat output that contains the port we're looking for.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(.*?:%i.)(.*)( [0-9]+)"), Port));
+		FRegexMatcher PidMatcher(PidMatcherPattern, NetStatResult);
+		if (PidMatcher.FindNext())
+		{
+			OutState = PidMatcher.GetCaptureGroup(2 /* Get the State of the process, which is the second group. */);
+			OutPid = PidMatcher.GetCaptureGroup(3 /* Get the PID, which is the third group. */);
+		}
+		else
+		{
+			bSuccess = false;
+			UE_LOG(LogSpatialCommandUtils, Error, TEXT("Failed to find PID of the process that is blocking %i port."), Port);
+		}
+	}
+	else
+	{
+		bSuccess = false;
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("Failed to find the process that is blocking required port. Error: %s"), *StdErr);
+	}
+
+	return bSuccess;
+}
+
+bool SpatialCommandUtils::GetProcessInfoFromPortMacOs(int32 Port, FString& OutPid, FString& OutState, FString& OutProcessName)
```

Comment:
overall the logic for both mac and windows are the same. afaics only the command and the regex are different. Can we merge this into one function or is the logic too different?